diggicamp/application/basics.py

```diff
@@ -57,30 +59,10 @@ def add_download(dgc: Diggicamp, folder_id: str, target: str, regex=None):
 
     if not dgc.conf.get('downloads'):
         dgc.conf.set('downloads', [entry])
-    else:
+    elif not any(map(lambda download_entry: download_entry[download_type] == download_id, dgc.conf.get('downloads'))):
```

Comment:
This is a little bit clunky to read, I would suggest:
``` python
not any((entry[download_type] == download_id for entry in dgc.conf.get('downloads')))
```

This just passes an iterator and expresses the `map` as an iterator comprehension. It should be about as fast as the list is still lazily evaluated. 

Alternatively, this approach might be even more readable (but is definitely slower for large inputs):

```python3
download_id not in [entry[download_type] for entry in dgc.conf.get('downloads')]
```

All these are untested, but I'm pretty sure they work.

Reply:
Committed the first proposal due to the best compromise between readability and performance.

Reply:
Awesome! Thanks!